src/ansys/sound/core/psychoacoustics/tonality_aures.py

```
@@ -0,0 +1,328 @@
+            )
+        self.__overlap = overlap
+
+    @property
+    def account_for_w1(self) -> bool:
+        """Specifies whether bandwidth weighting w1 should be taken into account or not.
+
+        Given the rather high line spacing (12.5 Hz), as specified in the reference paper of
+        Terhardt et al, the bandwidth weighting w1 is disabled by default. If the value is set to
+        `True`, then the tone bandwidth taken into account to calculate the weighting w1 is defined
+        as an X-dB bandwidth around the tone frequency, where X is the value set in the
+        :attr:`w1_threshold` attribute.
+        """
+        return self.__account_for_w1
+
+    @account_for_w1.setter
+    def account_for_w1(self, account_for_w1: bool):
+        """Set the account_for_w1."""
+        self.__account_for_w1 = account_for_w1
+
+    @property
+    def w1_threshold(self) -> float:
+        """Threshold in dB for the bandwidth weighting w1.
+
+        Defines the threshold value in dB for the tone bandwidth in the calculation of the
+        bandwidth weighting w1, if attribute :attr:`account_for_w1` is set to `True`. In this case,
+        the bandwidth is defined as an X-dB bandwidth around the tone frequency, where X is the
+        value set in this attribute.
+        """
+        return self.__w1_threshold
+
+    @w1_threshold.setter
+    def w1_threshold(self, w1_threshold: float):
+        """Set the w1_threshold."""
+        if w1_threshold < 0.0:
+            raise PyAnsysSoundException(
+                "Threshold for the bandwidth weighting must be greater than or equal to 0 dB."
+            )
+        self.__w1_threshold = w1_threshold
+
+    def process(self):
+        """Compute the Aures tonality."""
+        if self.signal is None:
+            raise PyAnsysSoundException(f"No input signal is set. Use {__class__.__name__}.signal.")
+
+        self.__operator.connect(0, self.signal)
+        self.__operator.connect(1, float(self.overlap))
```

@ansaminard I'm not sure regarding this cast, because it can "hide an error" by forcing a wrong parameter into a "good one" , what do you think ?

---

@a-bouth I feel it makes the life of the user easier, as this parameter is provided as a percentage.
Most often they will give round numbers, and I feel it is a bit of a pity when you get an error just
because you didn't add ".0" after your parameter.
(Also FYI I did the same in LevelOverTime, and OverallLevel)